### PR TITLE
validation apply new transformer

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?

to apply caffe related model with opencv api

## How was this patch tested?

unit test and model validation example

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

**Comment by wzhongyuan** [comment]
@SeaOfOcean  please help review, verified with example, running the PR job now

**Comment by wzhongyuan** [comment]
PR validation passed http://172.168.2.101:8080/job/BigDL-PR-V/789/

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/loadmodel/DatasetUtil.scala [hunk 1]
```diff
@@ -48,12 +50,14 @@ object AlexNetPreprocessor {
...
       // do not normalize the pixel values to [0, 1]
-    val transfomer = BytesToBGRImg(normalize = 1f, 256, 256) ->
-      BGRImgPixelNormalizer(means) -> BGRImgCropper(imageSize, imageSize, CropCenter) ->
-      BGRImgToSample(toRGB = false)
-    transfomer(dataSet)
+    val transfomer = PixelBytesToMat() -> Resize(256, 256) ->
+      PixelNormalizer(means.storage.array) -> CenterCrop(imageSize, imageSize) ->
+      MatToTensor[Float]() -> ImageFrameToSample[Float](targetKeys = Array(ImageFeature.label))
+    val imgFrame = data -> transfomer
+    val validImageFeatures = imgFrame.asInstanceOf[DistributedImageFrame].rdd.filter(_.isValid)
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/loadmodel/DatasetUtil.scala]: toDistributed()

Comment (SeaOfOcean) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/loadmodel/DatasetUtil.scala]: no need filter(_.isValid) here if properly processed?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/loadmodel/DatasetUtil.scala [hunk 2]
```diff
@@ -76,11 +80,13 @@ object InceptionPreprocessor {
...
-    val transfomer = BytesToBGRImg(normalize = 1f) ->
-      BGRImgCropper(imageSize, imageSize, CropCenter) ->
-      BGRImgNormalizer(123, 117, 104, 1, 1, 1) -> BGRImgToSample(toRGB = false)
-    transfomer(dataSet)
+    val data = DataSet.SeqFileFolder.filesToImageFrame(path, sc, 1000)
+    val transfomer = PixelBytesToMat() -> Resize(256, 256) ->
+      CenterCrop(imageSize, imageSize) -> ChannelNormalize(123, 117, 104) ->
+      MatToTensor[Float]() -> ImageFrameToSample[Float](targetKeys = Array(ImageFeature.label))
+    val imgFrame = transfomer(data)
+    val validImageFeatures = imgFrame.asInstanceOf[DistributedImageFrame].rdd.filter(_.isValid)
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/loadmodel/DatasetUtil.scala]: toDistributed()

Comment (SeaOfOcean) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/loadmodel/DatasetUtil.scala]: no need filter(_.isValid) here if properly processed

**Comment by wzhongyuan** [comment]
@SeaOfOcean  updated and verified

**Pull request merged**